app/controllers/patients_controller.rb

```diff
@@ -89,6 +89,18 @@ def data_entry_create
     end
   end
 
+  # allow ONLY admins to delete patients, and only then if patient does not have pledge_sent
+  def destroy
+    @patient = Patient.find params[:patient_id]
+    if current_user.admin? && !@patient.pledge_sent?
+      current_user.remove_patient(@patient)
+      redirect_to dashboard_path
+      flash[:notice] = "Patient successfully deleted"
+    else
+      flash[:alert] = "Errors prevented this patient from being deleted: #{patient.errors.full_messages.to_sentence}"
```

Comment:
Two things here:

a) The `patient.errors` is more for model level errors, whereas this is more on the order of 'you can't do that for some reason'. I'd suggest changing this message to something along the lines of 'Can't delete patients with pledges; please correct the patient record and try again.'

b) Add a `render :edit` after this, so it re-shows the page with the flash message (and so it tells future us what we should expect to happen).